src/app/services/filters.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { Subject } from 'rxjs';
+import { FiltersInterface } from '../interfaces/filters-interface';
+import { ProductInterface } from '../interfaces/product-interface';
+
+@Injectable()
+export class FiltersService {
+    private filters: FiltersInterface;
+    private handleFiltersSubject = new Subject<FiltersInterface>();
+
+    constructor() {
+    }
+
+    public handleFilters(data): void {
+        this.handleFiltersSubject.next(data);
+        this.handleFiltersSubject.subscribe((filters: FiltersInterface) => {
```

Comment:
Why do you need this `subsription` in service? This will cause memory leak.
You have to `subscribe` only at your components...

Reply:
Fixed